.github/workflows/build.yaml

```diff
@@ -0,0 +1,54 @@
+name: build
+on:
+  push:
+  pull_request:
+  schedule:  # nightly build
+    - cron: '0 0 * * *'
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: python -m pip install tox
+      - run: python -m tox -e mypy
```

Comment:
don't we already run mypy in pre-commit?

Reply:
We do, and it's great for catching most things. I added `mypy` to the tox+azure setup a while back when it passed under pre-commit but `mypy src/` failed.

I now think of the pre-commit run as a way letting `mypy` catch _most_ errors at commit-time, but `tox -e mypy` as "the true `mypy` run".

I don't know exactly how right or wrong that is, but it's how I reacted to `mypy src/` and `mypy src/**/*.py` being different.

Reply:
Ah, that's odd that those return different results.

I wonder if it makes sense to use a `local` hook in pre-commit so that there's parity between CI and local pre-commit runs, as in https://github.com/sloria/environs/blob/cbbe3ae9d970cbf4c495358077b6048e0f2476c8/.pre-commit-config.yaml#L22-L30

Also, it probably doesn't make sense to run mypy twice in CI. so keep this line and add

```yaml
ci:
  # mypy gets run in GitHub actions
  skip: [mypy]
```

to `.pre-commit-config.yaml`.

Reply:
Not sure this is relevant, but I have a project in which I don't want all pre-commit hooks to be run in CI, so I invoke them one by one in [tox.ini](https://github.com/BEMServer/bemserver-core/blob/master/tox.ini):

```py
[testenv:lint]
deps =
    pre-commit
skip_install = true
commands =
    pre-commit run pyupgrade --all-files --show-diff-on-failure
    pre-commit run black --all-files --show-diff-on-failure
    pre-commit run flake8 --all-files --show-diff-on-failure
```

I was reluctant at first but then realized that running all hooks was a shortcut but we might not always want to do that and calling them explicitly makes sense.

Reply:
i think we do want all the hooks to run in pre-commit.ci though right? mypy is just a special case

Reply:
`mypy` is definitely a (very?) special case. e.g. `black`, `flake8`, etc don't need app dependencies installed, but `mypy` does in order to pick up typing info from libraries. That's why I prefer `mypy` under `tox`, since it does the full install before running.

The more I've dealt with it, the more I find `mypy` under pre-commit weird. I might also experiment with a `local` hook which runs... `tox -e mypy`! I don't know yet. It's a bit unclear to me right now.

I'll add the skip for pre-commit.ci , but I don't want to hold up the basic switchover to github actions on details. So I'd like to punt on some of this stuff.